Cubical/Algebra/ZariskiLattice/Base.agda

```diff
@@ -485,3 +501,85 @@ module SmallZarLat (R' : CommRing ℓ) where
 
  ZL≃ZL' : ZL ≃ ZL'
  ZL≃ZL' = isoToEquiv IsoLarLatSmall
+
+
+module BasicOpens (R' : CommRing ℓ) where
+ open CommRingStr ⦃...⦄
+ open RingTheory (CommRing→Ring R')
+ open CommIdeal R'
+ open isCommIdeal
+
+ open ZarLat R'
+ open ZarLatUniversalProp R'
+ open IsZarMap
+
+ open Join ZariskiLattice
+ open IsBasis
+
+ private
+  R = fst R'
+  instance
+   _ = snd R'
+  ⟨_⟩ : {n : ℕ} → FinVec R n → CommIdeal
+  ⟨ V ⟩ = ⟨ V ⟩[ R' ]
+
+ BasicOpens : ℙ ZL
+ BasicOpens 𝔞 = (∃[ f ∈ R ] (D f ≡ 𝔞)) , isPropPropTrunc
+
+ BO : Type (ℓ-suc ℓ)
+ BO = Σ[ 𝔞 ∈ ZL ] (𝔞 ∈ₚ BasicOpens)
+
+ basicOpensAreBasis : IsBasis ZariskiLattice BasicOpens
+ contains1 basicOpensAreBasis = ∣ 1r , isZarMapD .pres1 ∣
+ ∧lClosed basicOpensAreBasis 𝔞 𝔟 = map2
+            λ (f , Df≡𝔞) (g , Dg≡𝔟) → (f · g) , isZarMapD .·≡∧ f g ∙ cong₂ (_∧z_) Df≡𝔞 Dg≡𝔟
+ ⋁Basis basicOpensAreBasis = elimProp (λ _ → isPropPropTrunc) Σhelper
+  where
+  Σhelper : (a : Σ[ n ∈ ℕ ] FinVec R n)
+          → ∃[ n ∈ ℕ ] Σ[ α ∈ FinVec ZL n ] (∀ i → α i ∈ₚ BasicOpens) × (⋁ α ≡ [ a ])
+  Σhelper (n , α) = ∣ n , (D ∘ α) , (λ i → ∣ α i , refl ∣) , path ∣
+   where
+   path : ⋁ (D ∘ α) ≡ [ n , α ]
+   path = funExt⁻ (cong fst ZLUniversalPropCorollary) _
+
+
+ -- The structure presheaf on BO
+ BOCat : Category (ℓ-suc ℓ) (ℓ-suc ℓ)
+ BOCat = ΣPropCat (DistLatticeCategory ZariskiLattice) BasicOpens
```

Comment:
One problem that maybe should be discussed before merging:
Using general machinery to define the structure sheaf through the universal property of localization,
we view the basic opens to as a category in the above way. This unfolds to them being a `JoinSemilatticeCategory`.

In the file [Cubical.Categories.DistLatticeSheaf](https://github.com/agda/cubical/blob/master/Cubical/Categories/DistLatticeSheaf.agda#L115), the basic opens are viewed as a `MeetSemilatticeCategory`.

The two definitions come from  the two ways of defining the order relation on a distributive lattice
(`x≤y` as `y=x\/y` vs `x=x/\y`) and they're thus equivalent but not definitionally equal.

I just thought maybe want make things uniform in one way or the other...

Reply:
Good catch! It's probably easier to change the DistLatticeSheaf? I can try to remember to do it when working on that file